dds/idl/typeobject_generator.cpp

```diff
@@ -519,18 +519,245 @@ operator<<(std::ostream& out, const OpenDDS::XTypes::TypeIdentifier& ti)
+  }
+}
+
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::AppliedVerbatimAnnotation& ann)
+{
+  return out
+    << "XTypes::AppliedVerbatimAnnotation("
+    << ann.placement << ", "
+    << ann.language << ", "
+    << ann.text
+    << ")";
+}
+
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::AppliedBuiltinTypeAnnotations& ann)
+{
+  return out
+    << "XTypes::AppliedBuiltinTypeAnnotations("
+    << ann.verbatim
+    << ")";
+}
+
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::NameHash& name_hash)
+{
+  for (size_t i = 0; i < sizeof name_hash; ++i) {
+    out << int(name_hash[i]);
+    if (i < sizeof name_hash - 1) {
+      out << ", ";
+    }
+  }
+  return out;
+}
+
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::AnnotationParameterValue& param_value)
+{
+  out << "XTypes::AnnotationParameterValue(";
+  switch (param_value.kind) {
+  case OpenDDS::XTypes::TK_BOOLEAN:
+    out << (param_value.boolean_value ? "true" : "false");
+    break;
+  case OpenDDS::XTypes::TK_BYTE:
+    out << "static_cast<ACE_CDR::Octet>(" << param_value.byte_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_INT16:
+    out << "static_cast<ACE_CDR::Short>(" << param_value.int16_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_UINT16:
+    out << "static_cast<ACE_CDR::UShort>(" << param_value.uint16_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_INT32:
+    out << "XTypes::TK_INT32, " << param_value.int32_value;
+    break;
+  case OpenDDS::XTypes::TK_UINT32:
+    out << "static_cast<ACE_CDR::ULong>(" << param_value.uint32_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_INT64:
+    out << "static_cast<ACE_CDR::LongLong>(" << param_value.int64_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_UINT64:
+    out << "static_cast<ACE_CDR::ULongLong>(" << param_value.uint64_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_FLOAT32:
+    out << "static_cast<ACE_CDR::Float>(" << param_value.float32_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_FLOAT64:
+    out << "static_cast<ACE_CDR::Double>(" << param_value.float64_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_FLOAT128:
+    out << "static_cast<ACE_CDR::LongDouble>(" << param_value.float128_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_CHAR8:
+    out << "static_cast<ACE_CDR::Char>(" << param_value.char_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_CHAR16:
+    out << "XTypes::AnnotationParameterValue::WCharValue(" << param_value.wchar_value << "))";
+    break;
+  case OpenDDS::XTypes::TK_ENUM:
+    out << "XTypes::AnnotationParameterValue::EnumValue(" << param_value.enumerated_value << "))";
+    break;
+  case OpenDDS::XTypes::TK_STRING8:
+    out << "String(" << param_value.string8_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_STRING16:
+    // Assuming each wide character can be converted to a single byte character.
```

Comment:
Create JIRA task to support wide characters.

Reply:
Has the Jira task been created?

Reply:
A task is created in JIRA (https://opendds.atlassian.net/browse/OPENDDS-339).